pandas/core/reshape/merge.py

```diff
@@ -1452,6 +1454,147 @@ def _validate(self, validate: str) -> None:
             raise ValueError("Not a valid argument for validate")
 
 
+def _chunks(lst, n):
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+_DEFAULT_LEFT_CHUNK_SIZE = _DEFAULT_RIGHT_CHUNK_SIZE = 1000
```

Comment:
This implicit chunking probably needs to be documented just in case a user passes an `on` callable that depends on the size of the data.

Reply:
handled this through new pandas options (`conditional_merge.<left|right>_chunk_size`) and noted these options in the `on` docstring